### ISSUE TITLE
workflow controller conflict if kubeflow installed 

### ISSUE THREAD

**Issue by Stevenyu1982** [argoproj/argo-workflows#6390]
## Summary 

I am testing both argo workflow and kubeflow now, so i have both installed in the same k8s cluster. when I try to submit job into the argo workflow which in the namespace argo,  the argo-server seems will talk to the workflow controller in kubeflow. 

this is very interesting, i am not sure how to override it.  here is what  looks like

in the namespace argo
k get pod                                                                                                                                                                           17:08:12
NAME                                   READY   STATUS              RESTARTS   AGE
argo-server-7577f8cbdf-mclp6           1/1     Running             2          29m
hello-world-w2clh                      0/2     ContainerCreating   0          115s
minio-644b476ff-76h7c                  1/1     Running             0          125m
postgres-59dd7646f4-xxfzj              1/1     Running             0          29m
workflow-controller-86f659b46b-xpjf9   1/1     Running             0          11m

and in the namespace kubeflow

k get pod -n kubeflow                                                                                                                                                               17:09:54
NAME                                                        READY   STATUS    RESTARTS   AGE
admission-webhook-deployment-f7547f56c-sl7q7                1/1     Running   2          16d
cache-deployer-deployment-d949f58fc-dt5xb                   2/2     Running   4          15d
cache-server-69c8b4485b-xlv8t                               2/2     Running   0          6d6h
centraldashboard-6d5df8dbb8-mh69f                           1/1     Running   1          16d
jupyter-web-app-deployment-7c8c98bc96-qgrlt                 1/1     Running   2          14d
katib-controller-6fbc9b4b67-7l9hk                           1/1     Running   1          15d
katib-db-manager-df996f75-7pshm                             1/1     Running   1          16d
katib-mysql-75f45688c5-lgvgr                                1/1     Running   1          16d
katib-ui-6bf9456f49-66sf5                                   1/1     Running   1          16d
kfserving-controller-manager-0                              2/2     Running   2          16d
kubeflow-pipelines-profile-controller-85f6b45949-fngmg      1/1     Running   0          6d6h
metacontroller-0                                            1/1     Running   0          6d6h
metadata-envoy-deployment-69794b47df-z8w5n                  1/1     Running   1          15d
metadata-grpc-deployment-564799bbf5-mdkp4                   2/2     Running   8          15d
metadata-writer-d6c4cc7bf-rncnz                             2/2     Running   5          15d
minio-56cdb4d65d-lbf4l                                      2/2     Running   2          15d
ml-pipeline-6f4b86c479-nbxvf                                2/2     Running   4          15d
ml-pipeline-persistenceagent-846446ff5-nl7mn                2/2     Running   2          15d
ml-pipeline-scheduledworkflow-64d4bc57ff-mhtrk              2/2     Running   2          15d
ml-pipeline-ui-549b449674-bzbm7                             2/2     Running   0          47h
ml-pipeline-viewer-crd-6b87f684f6-sbkzn                     2/2     Running   1          47h
ml-pipeline-visualizationserver-5bff865599-jrmwv            2/2     Running   0          6d6h
mpi-operator-7fc5b8bb7f-4c2q6                               1/1     Running   1          16d
mxnet-operator-5f88585f87-vtwhl                             1/1     Running   1          16d
mysql-7f4d8bcf6f-4vhrd                                      2/2     Running   0          6d6h
notebook-controller-deployment-664d6ddbc7-kshwv             1/1     Running   2          15d
profiles-deployment-7b499c6fbb-9759q                        2/2     Running   2          16d
pytorch-operator-b6ccd58f8-6jdvd                            2/2     Running   2          15d
tensorboard-controller-controller-manager-5b7cfd57f-9hj5x   3/3     Running   6          15d
tensorboards-web-app-deployment-75c7f59bf-q77vc             1/1     Running   1          15d
tf-job-operator-8587f66445-kmndp                            1/1     Running   0          6d6h
volumes-web-app-deployment-7cb949576b-vp48g                 1/1     Running   0          47h
workflow-controller-6548ff5d8c-ffbfj                        2/2     Running   6          15d
xgboost-operator-deployment-684fc5646f-j8nk6                1/1     Running   1          16d

From the namespace kubeflow , ran this command

 k logs workflow-controller-6548ff5d8c-ffbfj -n kubeflow

I am getting these logs, as you can see over here, the job has been scheduled in the argo workflow contorller in kuebflow. 

time="2021-07-21T09:04:31.181Z" level=info msg="Re-applying updates on latest version and retrying update" namespace=argo workflow=lovely-poochenheimer
time="2021-07-21T09:04:31.184Z" level=info msg="Failed to re-apply update: must never update completed workflows" namespace=argo workflow=lovely-poochenheimer
time="2021-07-21T09:04:31.184Z" level=info msg="archiving workflow" namespace=argo uid=90928a99-2b80-4cac-bf28-df47df6b80e5 workflow=lovely-poochenheimer
time="2021-07-21T09:04:31.191Z" level=info msg="archiving workflow" namespace=argo uid=90928a99-2b80-4cac-bf28-df47df6b80e5 workflow=lovely-poochenheimer
time="2021-07-21T09:04:31.192Z" level=info msg="Queueing Succeeded workflow argo/lovely-poochenheimer for delete in 5m0s"
time="2021-07-21T09:04:31.192Z" level=info msg="Queueing Succeeded workflow argo/lovely-poochenheimer for delete in 5m0s"
time="2021-07-21T09:05:03.240Z" level=info msg="Processing workflow" namespace=argo workflow=hello-world-4j6cj
time="2021-07-21T09:05:03.240Z" level=info msg="Updated phase  -> Running" namespace=argo workflow=hello-world-4j6cj
time="2021-07-21T09:05:03.241Z" level=info msg="Pod node hello-world-4j6cj initialized Pending" namespace=argo workflow=hello-world-4j6cj
time="2021-07-21T09:05:03.255Z" level=info msg="Created pod: hello-world-4j6cj (hello-world-4j6cj)" namespace=argo workflow=hello-world-4j6cj
time="2021-07-21T09:05:03.263Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=329826247 workflow=hello-world-4j6cj
time="2021-07-21T09:05:13.256Z" level=info msg="Processing workflow" namespace=argo workflow=hello-world-4j6cj
time="2021-07-21T09:07:52.648Z" level=info msg="Processing workflow" namespace=argo workflow=hello-world-w2clh
time="2021-07-21T09:07:52.648Z" level=info msg="Updated phase  -> Running" namespace=argo workflow=hello-world-w2clh
time="2021-07-21T09:07:52.649Z" level=info msg="Pod node hello-world-w2clh initialized Pending" namespace=argo workflow=hello-world-w2clh
time="2021-07-21T09:07:52.664Z" level=info msg="Created pod: hello-world-w2clh (hello-world-w2clh)" namespace=argo workflow=hello-world-w2clh
time="2021-07-21T09:07:52.672Z" level=info msg="Workflow update successful" namespace=argo phase=Running resourceVersion=329830001 workflow=hello-world-w2clh
time="2021-07-21T09:08:02.666Z" level=info msg="Processing workflow" namespace=argo workflow=hello-world-w2clh
time="2021-07-21T09:09:16.323Z" level=info msg="Alloc=8696 TotalAlloc=15137856 Sys=77248 NumGC=5235 Goroutines=180"
time="2021-07-21T09:09:32.000Z" level=info msg="Deleting TTL expired workflow 'argo/lovely-poochenheimer'"
time="2021-07-21T09:09:32.010Z" level=info msg="Workflow already deleted 'argo/lovely-poochenheimer'"

Does anyone know how to force the argo-server using the workflow-controller in the same namespace

**Comment by Stevenyu1982** [comment]
I have found the issue, 
this is because the workflow CRD , both argo and kubeflow will install the CRD, however when kubeflow install the CRD, it will add below labels.  so when apply the workflow jobs it will talk to workflow-controller in kubeflow namespace due this label. 

  labels:
    application-crd-id: kubeflow-pipelines
